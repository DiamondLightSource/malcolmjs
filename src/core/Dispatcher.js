/*
 * React.js Starter Kit
 * Copyright (c) 2014 Konstantin Tarkus (@koistya), KriaSoft LLC.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

import Flux from 'flux';
import PayloadSources from '../constants/PayloadSources';
import assign from 'react/lib/Object.assign';

/**
 * A singleton that operates as the central hub for application updates.
 * For more information visit https://facebook.github.io/flux/
 */
let Dispatcher = assign(new Flux.Dispatcher(), {

  /**
   * @param {object} action The details of the action, including the action's
   * type and additional data coming from the server.
   */
  handleServerAction(action) {
    var payload = {
      source: PayloadSources.SERVER_ACTION,
      action: action
    };
    this.dispatch(payload);
  },

  /**
   * @param {object} action The details of the action, including the action's
   * type and additional data coming from the view.
   */
  handleViewAction(action) {
    var payload = {
      source: PayloadSources.VIEW_ACTION,
      action: action
    };
    this.dispatch(payload);
  }

});

export default Dispatcher;
