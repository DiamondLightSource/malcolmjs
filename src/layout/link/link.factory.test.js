import { createShallow } from '@material-ui/core/test-utils';
import { DiagramEngine, PointModel, Toolkit } from 'storm-react-diagrams';
import LinkFactory from './link.factory';
import LinkModel from './link.model';

describe('LinkFactory', () => {
  let shallow;
  const linkFactory = new LinkFactory();

  beforeEach(() => {
    shallow = createShallow();
    Toolkit.TESTING = true;
  });

  it('generates react widget', () => {
    const link = new LinkModel();
    link.points = [
      new PointModel(link, { x: 0, y: 0 }),
      new PointModel(link, { x: 100, y: 0 }),
    ];
    link.sourcePort = {
      in: false,
    };
    link.theme = {
      palette: {
        background: { default: '#424242' },
        secondary: { main: '#ff0000' },
      },
    };

    const engine = new DiagramEngine();
    engine.nodesRendered = true;
    engine.registerLinkFactory(linkFactory);

    const wrapper = shallow(linkFactory.generateReactWidget(engine, link));
    expect(wrapper).toMatchSnapshot();
  });

  it('generates new instance', () => {
    const model = linkFactory.getNewInstance();
    expect(model).toBeInstanceOf(LinkModel);
  });
});
