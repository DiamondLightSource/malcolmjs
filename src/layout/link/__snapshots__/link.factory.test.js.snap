// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkFactory generates react widget 1`] = `
<g
  className="srd-default-link "
>
  <g
    key="link-0"
  >
    <path
      className="srd-default-link--highlight-path "
      d="M 0,0 C 50,0 50,0 100,0"
      onClick={[Function]}
      stroke="rgba(255,255,255,0.5)"
      strokeWidth={3}
    />
    <path
      className="srd-default-link--highlight-path "
      d="M 0,0 C 50,0 50,0 100,0"
      data-linkid="1"
      data-point={0}
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      stroke="rgba(255,255,255,0.5)"
      strokeLinecap="round"
      strokeOpacity={0}
      strokeWidth={20}
    />
  </g>
  <g
    key="point-5"
  >
    <circle
      className="point srd-default-link__point "
      cx={100}
      cy={0}
      r={5}
    />
    <circle
      className="point srd-default-link__point "
      cx={100}
      cy={0}
      data-id="5"
      data-linkid="1"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      opacity={0}
      r={15}
    />
  </g>
</g>
`;
